From 9b0cf35c363492482b3c865238a5c350f006946a Mon Sep 17 00:00:00 2001
From: Daniel Templeton <templedf@apache.org>
Date: Thu, 3 Nov 2016 14:03:28 -0700
Subject: [PATCH 2002/2848] YARN-4092 Addendum. Fixed UI redirection to print
 useful messages when both RMs are in standby
 mode. Contributed by Xuan Gong

Added lines lines missing from incomplete backport.

Change-Id: I55d606c903c4b9590753f83c82e7018856c2fff5
---
 .../apache/hadoop/yarn/client/TestRMFailover.java  |   10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-client/src/test/java/org/apache/hadoop/yarn/client/TestRMFailover.java b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-client/src/test/java/org/apache/hadoop/yarn/client/TestRMFailover.java
index 1e7c880..307e2d9 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-client/src/test/java/org/apache/hadoop/yarn/client/TestRMFailover.java
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-client/src/test/java/org/apache/hadoop/yarn/client/TestRMFailover.java
@@ -325,6 +325,16 @@ public void testRMWebAppRedirect() throws YarnException,
 
     redirectURL = getRedirectURL(rm2Url + "/proxy/" + fakeAppId);
     assertNull(redirectURL);
+
+    // transit the active RM to standby
+    // Both of RMs are in standby mode
+    getAdminService(0).transitionToStandby(req);
+    // RM2 is expected to send the httpRequest to itself.
+    // The Header Field: Refresh is expected to be set.
+    redirectURL = getRefreshURL(rm2Url);
+    assertTrue(redirectURL != null
+        && redirectURL.contains(YarnWebParams.NEXT_REFRESH_INTERVAL)
+        && redirectURL.contains(rm2Url));
   }
 
   // set up http connection with the given url and get the redirection url from the response
-- 
1.7.9.5

